### PR TITLE
Improve DesignWnd empty list prompts.

### PR DESCRIPTION
**Pull request by LGM-Doyle**
This PR should follow #1972.

It allows a design's browse text  to be Obsolete and Unavailable at the same time.

This PR improves several of the empty list prompts in the `DesignWnd`:
- adds the "Toggle the Obsolete, Available and Unavailable filters to show more" message to all lists that could be empty because the availabilities filter is removing all of the item.
- adds "Neither the Saved Designs nor the Default Designs were added to the empire at game start.  Options to add these designs at game start can be changed in Options->Other.  Add the Saved Designs with the context menu of any design under the Saved tab." when no designs are imported into the Finished Designs.  This was reported as an issue in http://freeorion.org/forum/viewtopic.php?f=28&t=10667&sid=580f79fb27c603086f7fd7d7801a40fd.

### CODE REVIEW

**Comment by geoffthemedio** [comment]
rebase needed?

**Comment by LGM-Doyle** [comment]
rebased.

File: UI/DesignWnd.cpp [hunk 1]
```diff
@@ -905,39 +905,41 @@ namespace {
...
     class AvailabilityManager {
-    private:
-        // A tuple of the toogle state of the 3-tuple of coupled
-        // availability filters in the GUI:
-        // Obsolete, Available and Unavailable
-        std::tuple<bool, bool, bool> m_availabilities;
-
     public:
+        // DisplayedAvailabiltyType is indexed by Availability::Enum
+        using DisplayedAvailabilty = std::tuple<bool, bool, bool>;
```
Comment (geoffthemedio) [inline UI/DesignWnd.cpp:910]: should probably be `DisplayedAvailabilities` ?

at least ility instead of ilty

File: UI/DesignWnd.cpp [hunk 2]
```diff
@@ -905,39 +905,41 @@ namespace {
...
             AvailabiltyManager and that the X is \p available and \p obsolete,
             return the displayed state of the X. Return none if the X should
             not be displayed. */
-        boost::optional<Availability::Enum> DisplayedXAvailability(bool available, bool obsolete) const;
+        boost::optional<DisplayedAvailabilty> DisplayedXAvailability(bool available, bool obsolete) const;
 
+        // A tuple of the toogle state of the 3-tuple of coupled
+        // availability filters in the GUI:
+        // Obsolete, Available and Unavailable
+        std::tuple<bool, bool, bool> m_availabilities;
```
Comment (geoffthemedio) [inline UI/DesignWnd.cpp:942]: probably use the typedef discussed above here...

Comment (geoffthemedio) [reply UI/DesignWnd.cpp:942]: alternatively, make it an `map<Availability::Enum, bool>` ? would seem to simplify the data access without needing C++17 features.

File: UI/DesignWnd.cpp [hunk 3]
```diff
@@ -2656,6 +2663,26 @@ void MonstersListBox::PopulateCore() {
 }
 
 
+void BasesListBox::ResetEmptyListPrompt() {
+    SetEmptyPromptText(UserString("ALL_AVAILABILITY_FILTERS_BLOCKING_PROMPT"));
```
Comment (geoffthemedio) [inline UI/DesignWnd.cpp:2667]: prefer `{ ... }` single line

**Comment by geoffthemedio** [comment]
Otherwise works and OK.

**Pull request merged**